src/getCurrentConnectionInfo.ts

```diff
@@ -0,0 +1,27 @@
+export type ConnectionInfo = {
+  connectionIndex: number;
+
+  /** a string like "designer" | "explorer" | "studio" */
+  product: string;
+};
+
+/** returns info on the current Portal/Cloud connection */
+export const getCurrentConnectionInfo = (
+  window: Window
+): ConnectionInfo | null => {
+  const urlPathRegex = /\/u\/(?<index>[0-9]+)\/(?<product>.+?)\//;
+  const currentMatch = window.location.pathname.match(urlPathRegex);
+  if (
+    currentMatch &&
+    currentMatch.groups &&
```

Comment:
I think if there is a match the `groups` should not be null, and `groups.index` should always be set too

Reply:
typescript thinks groups can be undefined, but i can use the ?. operator:
[`7004d97` (#8)](https://github.com/stardog-union/portal-sdk/pull/8/commits/7004d97663032ef7dae6f804a2521c85eedba473)